rmc/src/handler.rs

```diff
@@ -127,3 +129,223 @@ impl<H: Signable + Hash + Eq + Clone + Debug, MK: MultiKeychain> Handler<H, MK>
         )
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::{
+        handler::{Error, OnStartRmcResponse},
+        Handler,
+    };
+    use aleph_bft_crypto::{NodeCount, NodeIndex, PartiallyMultisigned, Signed};
+    use aleph_bft_mock::{BadSigning, Keychain, Signable};
+
+    fn apply_signatures(
+        handler: &mut Handler<Signable, Keychain>,
+        hash: &Signable,
+        count: NodeCount,
+        nodes: impl Iterator<Item = NodeIndex>,
+    ) {
+        for i in nodes {
+            let keychain_i = Keychain::new(count, i);
+            let signed_hash = Signed::sign_with_index(hash.clone(), &keychain_i);
+            handler
+                .on_signed_hash(signed_hash.clone().into_unchecked())
+                .expect("the signatures should be correct");
+        }
+    }
+
+    fn apply_signatures_and_get_multisigned(
+        handler: &mut Handler<Signable, Keychain>,
+        hash: &Signable,
+        count: NodeCount,
+        nodes: impl Iterator<Item = NodeIndex>,
+    ) -> Option<PartiallyMultisigned<Signable, Keychain>> {
+        let mut multisigned = None;
+        for i in nodes {
+            let keychain_i = Keychain::new(count, i);
+            let signed_hash = Signed::sign_with_index(hash.clone(), &keychain_i);
+            handler
+                .on_signed_hash(signed_hash.clone().into_unchecked())
+                .expect("the signatures should be correct");
+            multisigned = match multisigned {
+                None => Some(signed_hash.into_partially_multisigned(&keychain_i)),
+                Some(ms) => Some(ms.add_signature(signed_hash, &keychain_i)),
+            }
+        }
+        multisigned
+    }
+
+    #[test]
+    fn on_start_rmc_before_exceeding_quorum_returns_signed() {
+        let hash: Signable = "13".into();
+        let keychain = Keychain::new(7.into(), 0.into());
+        let mut handler = Handler::new(keychain);
+        let expected = Signed::sign_with_index(hash.clone(), &keychain);
+        assert_eq!(
+            handler.on_start_rmc(hash),
+            OnStartRmcResponse::SignedHash(expected)
+        );
+    }
+
+    #[test]
+    fn on_start_rmc_exceeding_quorum_returns_multisigned() {
```

Comment:
s/exceeding/reaching, same in the comment below and the other test also about that, should be clearer